### ISSUE TITLE
Cite software dependencies in spectre DOI

### ISSUE THREAD

**Issue by nilsvu** [sxs-collaboration/spectre#3345]
We should add references to spectre's dependencies to the DOI. They can be added as Zenodo's "References" fields.

**Comment by nilsdeppe** [comment]
Should we also add a list of "If you ran the code, you likely used these packages and should cite them in this way" to the dox somewhere?

**Comment by nilsvu** [comment]
Sure. In terms of machine-readable information I think the info should be in our [Metadata.yaml](https://github.com/sxs-collaboration/spectre/blob/develop/Metadata.yaml) file. From that we auto-generate both the [CITATION.cff](https://github.com/sxs-collaboration/spectre/blob/develop/CITATION.cff) file (see format specification [here](https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md#referencing-other-work)) and the info we push to Zenodo.

**Comment by nilsdeppe** [comment]
:+1: that sounds great :)

**Comment by nilsdeppe** [comment]
I think this is done now, right @nilsvu ?

**Comment by nilsvu** [comment]
Yes they get synced from the metadata file in our repo, but we should update that with dependencies we have added (like xsimd, Kokkos).
